controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -510,6 +510,26 @@ func (r *BareMetalHostReconciler) actionUnmanaged(prov provisioner.Provisioner,
 	return actionContinue{unmanagedRetryDelay}
 }
 
+func (r *BareMetalHostReconciler) actionDetached(prov provisioner.Provisioner, info *reconcileInfo) actionResult {
+	if info.host.OperationalStatus() != metal3v1alpha1.OperationalStatusDetached {
+		provResult, err := prov.Detach()
+		if err != nil {
+			return actionError{errors.Wrap(err, "failed to detach")}
+		}
+		if provResult.ErrorMessage != "" {
+			return recordActionFailure(info, metal3v1alpha1.DetachError, provResult.ErrorMessage)
```

Comment:
We need to clear this error again somewhere - both the ErrorType/ErrorMessage the next time we call Detach without getting an ErrorMessage back, and the ErrorCount once we have successfully detached.

Reply:
Fixed in the latest version